File: lib/simple_pvr/database_schedule_reader.rb
require File.dirname(__FILE__) + '/pvr_initializer'
require File.dirname(__FILE__) + '/model/database_initializer'
require File.dirname(__FILE__) + '/recording_planner'

module SimplePvr
  class DatabaseScheduleReader
    def self.read
      @recording_planner = RecordingPlanner.new
      
      Model::Schedule.all.each do |schedule|
        @recording_planner.specification(title: schedule.title, channel: schedule.channel)
      end
      
      @recording_planner.finish
    end
  end
end

File: python/simple_pvr/database_schedule_reader.py
from .server import db

class DatabaseScheduleReader():

    def __init__(self):
        from .master_import import RecordingPlanner

        self._recording_planner = RecordingPlanner()

    def read(self):
        from .master_import import Schedule

        self._recording_planner._recordings = []
        for schedule in db.session.query(Schedule).all():
            self._recording_planner.specification(schedule.title, schedule.channel, schedule.start_time)
        
        self._recording_planner.finish()
